### DISCUSSION TITLE
Spring Cloud Data Flow context propagation from Spring boot service

### DISCUSSION THREAD

**Discussion by dimitri995** [open-telemetry/opentelemetry-java-instrumentation#7045] [Q&A] [answered]
Hello,

I would want to propagate context from spring boot application to a spring-cloud-task. We use SCDF to deploy & run the task, how can I pass the traceID / context in order to have the same traceparentId?

![image](https://user-images.githubusercontent.com/66120155/200029670-67d3b775-0e20-4346-a0b1-c50e2223e56f.png)

I don't use the java agent.

Thanks for your time,

Regards,

Dimitri

**Comment by trask** [comment, marked as answer]
hi @dimitri995, I don't think there's any auto- or library- instrumentation to support this currently, so you would need to propagate the trace context manually. for some details, check out https://opentelemetry.io/docs/instrumentation/java/manual/#context-propagation

**Reply by dimitri995** [reply]
Hi @trask ,

Thank you for your answer. But I can't propagate the context via headers. I can pass the context by an argument and I get this argument (context) in TaskService. But my question is how can I set a new context in TaskService (with context passed from SpringBoot Service as argument) ?
Context is immuable, so I can't set a new context...

I pass the actual spanContext from SpringBoot application: 
![image](https://user-images.githubusercontent.com/66120155/200577334-0aa89cb1-fb44-4424-a4f1-43583f0c4e35.png)


From SCDF : 
![image](https://user-images.githubusercontent.com/66120155/200576736-a9181ce5-1af3-4bb6-98d9-6d78d1211abc.png)

I would want to have just one traceID. Is it possible if I get the spanContext from the remote Service ? How can I do this ? 

Thanks for your time,

Regards,

**Reply by dimitri995** [reply]
Hi,
I found the way to propagate the context manually with this article https://github.com/open-telemetry/opentelemetry-java/discussions/4668.

I need to pass traceId & SpanID as argument in SCDF in order to get it into ServiceB.

Thank you for your time,

Regards,

Dimitri

**Reply by dimitri995** [reply]
Hi @trask,

I have one question.

Is it possible to force/insure to export Span before the shutdown of application ?

Because when I execute my application task from SCDF. Span are not exported. But when I execute this one in local, span are exported.

Regards,

Dimitri

**Reply by dimitri995** [reply]
I found a solution by using ForceFlush with SdkTracerProvider.forceFlux()

![image](https://user-images.githubusercontent.com/66120155/200840372-a6dfd341-39ae-4ee4-a812-977ff8fc0ba1.png)


![image](https://user-images.githubusercontent.com/66120155/200835885-4db12edd-7bd1-4f6e-8769-2335b410e0c5.png)
